lib/utilities.js

```diff
@@ -229,7 +231,48 @@ async function connectPort (udid, port, socket = null) {
   }
 }
 
+/**
+ * @typedef {
+ * import('./imagemounter/utils/list_developer_image').ImageFromGithubRepo
+ * | import('./imagemounter/utils/list_developer_image').ImagePath
+ * } MountOpts Local path or github repo
+ */
+
+/**
+ * mount developer image based on device version and local storage.
+ * @param {string} udid
+ * @param {MountOpts} opts local image and signature files or github repo option.
+ * @throws If opts error or mount faile
+ */
+async function mountDeveloperImage(udid, opts) {
+  const osVersion = await getOSVersion(udid);
+  let { developerImage, developerImageSignature } = opts;
+  // Not a local path in `ImagePath` format, try to use github repo
+  if (!developerImage || !developerImageSignature) {
+    // Not containing any github setting, invalid opts
+    if (!opts.repo) {
+      throw new TypeError(`Mount option should contains local path or github repo, got ${JSON.stringify(opts)} instead`);
+    }
+    try {
+      ({ developerImage, developerImageSignature } = await findDeveloperImage(osVersion, opts));
+    } catch (e) {
+      log.error(`Failed on fetching image for ${udid} with iOS ${osVersion}`);
+      throw e;
+    }
+  }
+  const imageMountService = await startImageMounterService(udid);
+  try {
+    await imageMountService.mount(developerImage, developerImageSignature);
+  } catch (e) {
+    log.error(`Failed to mount image for ${udid} with iOS ${osVersion} with ${JSON.stringify({ developerImage, developerImageSignature })}`);
```

Comment:
should the error be rethrown?

Reply:
yes, this method is no longer returning boolean